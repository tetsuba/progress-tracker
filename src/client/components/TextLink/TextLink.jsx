import React from 'react'

export default function TextLink({eventHandler, children}) {
    const styles = {color: '#007bff', cursor: 'pointer'};
    return (
        <span
            style={styles}
            id="forgotPasswordButton"
            onClick={eventHandler}
        >
            { children }
        </span>
    )
}
