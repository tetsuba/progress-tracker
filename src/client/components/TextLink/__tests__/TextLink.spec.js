import { testRenderer } from "../../../../test/testHelper";
import TextLink from "../TextLink";


describe('<Textlink>', () => {
    describe('@Render', () => {
        const props = {
            eventHandler: jest.fn(),
            children: ['This is a text link'],
        };

        it('should render component', () => {
            const wrapper = testRenderer(TextLink, props);
            expect(wrapper).toMatchSnapshot()
        });
    })

    describe('@Event', () => {
        const props = {
            eventHandler: jest.fn(),
            children: ['This is a text link'],
        };

        it('should call eventHandler on event click', () => {
            const wrapper = testRenderer(TextLink, props);
            wrapper.prop('onClick')();
            expect(props.eventHandler).toHaveBeenCalledTimes(1)
        });
    })
});