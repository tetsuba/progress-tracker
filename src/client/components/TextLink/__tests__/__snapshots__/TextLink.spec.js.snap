// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Textlink> @Render should render component 1`] = `
<span
  id="forgotPasswordButton"
  onClick={[MockFunction]}
  style={
    Object {
      "color": "#007bff",
      "cursor": "pointer",
    }
  }
>
  This is a text link
</span>
`;
